presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeCtasQueries.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createLineitem;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createNation;
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.createOrders;
+
+public abstract class AbstractTestNativeCtasQueries
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected void createTables()
+    {
+        QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner();
+        createLineitem(queryRunner);
+        createOrders(queryRunner);
+        createNation(queryRunner);
+    }
+
+    @Test
+    public void testCreateTableAsSelect()
+    {
+        assertUpdate("CREATE TABLE IF NOT EXISTS test_ctas AS SELECT name, regionkey FROM nation", "SELECT count(*) FROM nation");
+        assertTableColumnNames("test_ctas", "name", "regionkey");
+        assertUpdate("DROP TABLE test_ctas");
+        //Some connectors support CREATE TABLE AS but not the ordinary CREATE TABLE. Let's test CTAS IF NOT EXISTS with a table that is guaranteed to exist.
+        assertUpdate("CREATE TABLE IF NOT EXISTS nation AS SELECT orderkey, discount FROM lineitem", 0);
+        assertTableColumnNames("nation", "nationkey", "name", "regionkey", "comment");
+
+        assertCreateTableAsSelect(
+                "test_select",
+                "SELECT orderdate, orderkey, totalprice FROM orders",
+                "SELECT count(*) FROM orders");
+
+        assertCreateTableAsSelect(
+                "test_group",
+                "SELECT orderstatus, sum(totalprice) x FROM orders GROUP BY orderstatus",
+                "SELECT count(DISTINCT orderstatus) FROM orders");
+
+        assertCreateTableAsSelect(
+                "test_join",
+                "SELECT count(*) x FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey",
+                "SELECT CAST(1 as BIGINT)");
+
+        assertCreateTableAsSelect(
+                "test_limit",
+                "SELECT orderstatus FROM orders ORDER BY orderstatus LIMIT 10",
+                "SELECT CAST(10 as BIGINT)");
+
+        assertCreateTableAsSelect(
+                "test_unicode",
+                "SELECT '\u2603' unicode",
+                "SELECT CAST(1 as BIGINT)");
+
+        assertCreateTableAsSelect(
+                "test_with_data",
+                "SELECT * FROM orders WITH DATA",
```

Comment:
This is again a coordinator path as this is the default. Let's remove this test.